test/should_pass/binary_in_union.erl

```diff
@@ -0,0 +1,19 @@
+-module(binary_in_union).
+-export([
+         iodata_binary/0,
+         bitstring_fun/0,
+         nested_bitstrings/0
+        ]).
+
+-spec iodata_binary() -> iodata().
+iodata_binary() ->
+    <<<<"A">> || _ <- lists:seq(1, 10)>>.
+
+-spec bitstring_fun() -> <<_:7, _:_*3>> | string().
+bitstring_fun() ->
+    << <<1:N>> || N <- lists:seq(3, 12)>>.
```

Comment:
This shouldn't type check. The value is not of the expected type.

Reply:
Hmmm... I think I just saw why it shouldn't: https://github.com/josefs/Gradualizer/blob/c31c2326ad6b9674cefdb5c202f036bd86a150ce/src/typechecker.erl#L2776 There, it's pretty much accepting any sizes of M and N and making them simply 0 and 1, which will be compatible with everything else :slightly_frowning_face:

Reply:
Right... Only the word TODO is missing in the comment.

Your test case can be put under `test/known_problems/should_fail/`.

If the fixed bit size is zero though (e.g. a type like `<<_:_*3>>`) we are able to handle it.